test/status_file.c

```diff
@@ -0,0 +1,366 @@
+[broken\n\
+";
+	pathname = write_tmp_file(fixture->tmpdir, "existing_config.conf", cfg_file, NULL);
+	g_assert_nonnull(pathname);
+
+	r_context_conf()->configpath = g_strdup("test/test-global.conf");
+	r_context()->config->statusfile_path = pathname;
+	r_context()->system_status->boot_id = g_strdup("e02a2afe-cf45-4d50-a3f3-c223ca0f480a");
+
+	g_message("EXPECT FORM NOW!");
+
+	g_test_expect_message(G_LOG_DOMAIN, G_LOG_LEVEL_WARNING, "Failed to load status file: Key file contains line *");
+	g_test_expect_message(G_LOG_DOMAIN, G_LOG_LEVEL_WARNING, "Will move status file to * and re-create it.");
+
+	/* assert error-free saving*/
+	res = r_system_status_save(&ierror);
+	g_assert_no_error(ierror);
+	g_assert_true(res);
+
+	/* re-load key file for asserting content */
+	keyfile = g_key_file_new();
+	res = g_key_file_load_from_file(keyfile, pathname, G_KEY_FILE_NONE, &ierror);
+	g_assert_no_error(ierror);
+	g_assert_true(res);
+
+	/* assert loaded key file contains only newly added [system] group */
+	groups = g_key_file_get_groups(keyfile, &num_groups);
+	g_assert_cmpint(num_groups, ==, 1);
+	g_assert_true(g_strv_contains((const gchar * const *)groups, "system"));
+}
+
+int main(int argc, char *argv[])
+{
+	setlocale(LC_ALL, "C");
+
+	g_test_init(&argc, &argv, NULL);
+	/* Tests for slot status only */
+	g_test_add_func("/status-file/slot-status/read", status_file_test_read_slot_status);
+	g_test_add_func("/status-file/slot_status/write-read", status_file_test_write_slot_status);
+	g_test_add("/status-file/slot-staus/global", ConfigFileFixture, NULL,
+			status_file_fixture_set_up_global, status_file_test_global_slot_status,
+			status_file_fixture_tear_down);
+	/* Tests for system status only */
+	g_test_add("/status-file/system-status/load", ConfigFileFixture, NULL,
+			status_file_fixture_set_up_global,
+			status_file_test_load_system_status,
+			status_file_fixture_tear_down);
+	g_test_add("/status-file/system-status/load-broken", ConfigFileFixture, NULL,
+			status_file_fixture_set_up_global,
+			status_file_test_load_broken,
+			status_file_fixture_tear_down);
+	g_test_add("/status-file/system-status/save", ConfigFileFixture, NULL,
+			status_file_fixture_set_up_global,
+			status_file_test_save_system_status,
+			status_file_fixture_tear_down);
+	g_test_add("/status-file/system-status/save-per-slot", ConfigFileFixture, NULL,
+			status_file_fixture_set_up_global,
+			status_file_test_save_system_status_per_slot,
+			status_file_fixture_tear_down);
+	/* Combined tests */
+	g_test_add("/status-file/save-system-status-existing-slot-status", ConfigFileFixture, NULL,
+			status_file_fixture_set_up_global,
+			status_file_test_save_system_status_existing_slot_status,
+			status_file_fixture_tear_down);
+	g_test_add("/status-file/save-system-status-broken-existing-slot-status", ConfigFileFixture, NULL,
+			status_file_fixture_set_up_global,
+			status_file_test_save_system_status_broken_existing_slot_status,
+			status_file_fixture_tear_down);
```

Comment:
Do we need the other side (save slot status in a file with existing system status) as well?

Reply:
Thought about this, too. Since it uses the same helper it should not fail the other way round, but, maybe better safe than sorry. Will add a test for this, too.